Python/ceval.c

```diff
@@ -2182,6 +2191,30 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             DISPATCH();
         }
 
+        TARGET(BINARY_ADD_CACHED) {
+            SpecializedCacheEntry *caches = GET_CACHE();
+            _PyAdaptiveEntry *cache0 = &caches[0].adaptive;
+            PyObject *left = SECOND();
+            PyObject *right = TOP();
+            DEOPT_IF(Py_TYPE(left)->tp_version_tag != cache0->left_version, BINARY_ADD);
+            DEOPT_IF(Py_TYPE(right)->tp_version_tag != cache0->right_version, BINARY_ADD);
+            record_cache_hit(cache0);
+            binaryfunc function = caches[-1].func_ptr.function;
+            PyObject *sum = function(left, right);
+            if (sum == Py_NotImplemented) {
+                Py_DECREF(sum);
+                sum = NULL;
```

Comment:
You may want to consider replacing this with
```suggestion
                Py_SETREF(sum, NULL);
```

Reply:
Or:

```suggestion
                Py_CLEAR(sum);
```